File: src/main/java/info/szadkowski/contact/properties/TemplateProperties.java
package info.szadkowski.contact.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.io.Resource;

@Configuration
@ConfigurationProperties(prefix = "template")
@Data
public class TemplateProperties {
  private Resource resource;
}

File: src/main/kotlin/info/szadkowski/contact/properties/TemplateProperties.kt
package info.szadkowski.contact.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding
import org.springframework.core.io.Resource

@ConstructorBinding
@ConfigurationProperties(prefix = "template")
data class TemplateProperties(
    val resource: Resource
)
